Lib/pdb.py

```diff
@@ -1719,10 +1753,24 @@ def _help_message_from_doc(self, doc):
             elif i < usage_end:
                 prefix = "       "
             else:
+                if usage_only:
+                    break
                 prefix = ""
             formatted.append(indent + prefix + line)
         return "\n".join(formatted)
 
+    def _print_invalid_arg(self, arg):
+        """Return the usage string for a function."""
+
+        self.error(f"Invalid argument: {arg}")
```

Comment:
Maybe consider using `{arg!r}` to avoid confusion when `arg` contains spaces

Reply:
`!r` simply calls `repr()` for `arg` and as `arg` is `str`, it should have the identical behavior. Not sure what you are trying to achieve here, could you give an example where `!r` gives a different result?

Reply:
Oh I was wrong about `repr(str)` returns indentical result, don't know what I was thinking. `repr()` actually returns a worse result:

```python
(Pdb) r '"'
*** Invalid argument: '\'"\''
      Usage: r(eturn)
(Pdb) 
```

`{args}` prints exactly what user types and that is more intuitive.

Reply:
I usually use `repr` to add quotes. Didn't realize that there was additional escaping happening. The motivation here is to make `arg` appear as a single value. This really is just a nit, so...

Reply:
Actually, taking a look at the rest of the code, I realized people have different ideas about this. The current implementation contains a mix of `%r` and `%s` and different efforts to "contain" the argument with `'` or `()`. Some, however, is untouched. It might be beneficial if we could unify them - `%r` has the escaping side effect and we should use explicit quotes if possible.